weave/wandb_interface/wandb_artifact_pusher.py

```diff
@@ -46,17 +47,25 @@ def write_artifact_to_wandb(
 
     if _lite_run is None:
         lite_run = InMemoryLazyLiteRun(
-            entity_name, project_name, group="weave_artifact_pushers", _hide_in_wb=True
+            entity_name,
+            project_name,
+            group="weave_artifact_pushers",
+            _hide_in_wb=True,
+            _optimize_for_artifact_commit=True,
         )
     else:
         lite_run = _lite_run
 
+    entity_name = lite_run.run.entity or entity_name
```

Comment:
shouldn't entity_name and project_name always be valid here?

Reply:
Calling `.run` here kicks off the UpsertProject mutation. This code path is actually used when publishing brand new boards / creating brand new artifacts under projects don't yet exist. If we don't kick off the UpsertProject here, we have a bunch of tests that fail at the next line (CreateArtifactType) since the project doesn't exist. Which is why I assume the previous code called `lite_run.run.entity` and `lite_run.run.project`. 

I'm not a fan of this happening implicitly. So I'm actually going to place a `upsert_project` method on the LiteRun class that we can call explicitly so I can change this back to use the passed-in entity and project names.